Reduce gap between steps in SpacingSizesControl, add animation, remove first/last marks

jameskoster opened the pull request:
## What?
Refine the appearance of `SpacingSizesControl`.

## Why?
The white space between options makes the control feel heavy and noisy, drawing more attention than is warranted. This PR reduces it slightly.

Animation on the value / handle add polish and underline the nature of the range.

## Before

https://github.com/user-attachments/assets/6c44b736-1b1e-45f9-9851-057c3121d364



## After



https://github.com/user-attachments/assets/cb7f5cc7-655d-4907-ab7b-1661dd2c721d

jameskoster committed:
Animate track and handle

jameskoster committed:
Mark styling

bgardner commented:
Looks good and tests well for me! (Padding, Margin, Block Spacing)

richtabor commented:
I'm not feeling it to be honest. It's more disorienting seeing the steps poke out a bit, especially when the control has a value.  

![CleanShot 2024-07-22 at 16 04 38](https://github.com/user-attachments/assets/e2a660bb-ed30-483f-ae11-96a4a24f7d5b)

jameskoster committed:
Inline dot markers

jameskoster committed:
Lines

richtabor commented:
I appreciate the different explorations, but I don't think any are a big enough change/win to necessarily warrant the change. 

Perhaps just reducing the gap between sections to 2px would tighten this up a bit? 

![CleanShot 2024-07-24 at 12 28 42](https://github.com/user-attachments/assets/56fee7bb-fb06-45b0-9c08-f2f0126deb7d)

jameskoster commented:
Yes that might help.

jameskoster committed:
Mark styling

jameskoster commented:
I reduced the space to `1.5px` (a value we use elsewhere). `2px` could work too though, no strong opinion. I left the animation as that still feels like a nice enhancement to me.

Let me know what you think.

<img width="280" alt="Screenshot 2024-07-24 at 17 36 44" src="https://github.com/user-attachments/assets/e207834f-3424-407a-aad7-0fc5a830511f">

richtabor commented:
I lean towards 2px; Feels like a good balance, accounting for the general density of the other inspector controls. 

Animation looks nice. 

| 1.5px  | 2px |
| ------------- | ------------- |
|![CleanShot 2024-07-27 at 11 02 15](https://github.com/user-attachments/assets/eb2b2d0f-b8c3-46b5-b69e-02415add9a44)|![CleanShot 2024-07-27 at 11 01 49](https://github.com/user-attachments/assets/aafc1818-9ebe-4d7e-a0f0-057c024c312d)|

bgardner commented:
I prefer the 2px spacing, FWIW.

jameskoster committed:
Update spacing

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/style.scss:
```diff
@@ -21,14 +33,16 @@
 	.components-range-control__marks {
 		margin-top: 17px;
 
+		:last-child,
 		:first-child {
 			display: none;
 		}
```
Good catch on this.
<details><summary>Alternatively, I think it’d be nice if the component didn’t render the first or last marks to begin with. Then this ruleset can be removed.</summary>

```diff
diff --git a/packages/block-editor/src/components/spacing-sizes-control/input-controls/spacing-input-control.js b/packages/block-editor/src/components/spacing-sizes-control/input-controls/spacing-input-control.js
index 6605d2de50..8e20942dd8 100644
--- a/packages/block-editor/src/components/spacing-sizes-control/input-controls/spacing-input-control.js
+++ b/packages/block-editor/src/components/spacing-sizes-control/input-controls/spacing-input-control.js
@@ -193,10 +193,12 @@ export default function SpacingInputControl( {
 		name: size.name,
 	} ) );
 
-	const marks = spacingSizes.map( ( _newValue, index ) => ( {
-		value: index,
-		label: undefined,
-	} ) );
+	const marks = spacingSizes
+		.slice( 1, spacingSizes.length - 1 )
+		.map( ( _newValue, index ) => ( {
+			value: index + 1,
+			label: undefined,
+		} ) );
 
 	const sideLabel =
 		ALL_SIDES.includes( side ) && showSideInLabel ? LABELS[ side ] : '';

```
</details>

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/style.scss:
```diff
@@ -21,14 +33,16 @@
 
 	.components-range-control__mark {
 		height: $grid-unit-05;
```
It’s quite minute (especially now with their reduced width), but I believe the base positioning of the marks don’t account for their width so this should place them more accurately.
```suggestion
	.components-range-control__mark {
		transform: translateX(-50%);
		height: $grid-unit-05;
```

stokesman reviewed (commented):
I liked the Lines variation https://github.com/WordPress/gutenberg/pull/63803#issuecomment-2244882015 because the darker mark color feels more like a divot and doesn’t visually break the bar apart. Also, it makes the control less dependent on its background. Right now, if used on a background other than white the styles have to be overridden to make the marks match that background. Not that I feel too strongly on those points.

richtabor reviewed (approved):
LGTM

jameskoster committed:
mark position

jameskoster committed:
Do not render first/last marks

jameskoster commented:
Thanks for the feedback @stokesman, I addressed your points 👍 

> doesn’t visually break the bar apart

This bugs me too, but it seems to be quite subjective, so a detail to explore in isolation separately.
